src/rehype-filter.js

```diff
@@ -0,0 +1,46 @@
+const visit = require('unist-util-visit')
+
+const splice = [].splice
+
+module.exports = rehypeFilter
+
+function rehypeFilter(options) {
+  if (options.allowedElements && options.disallowedElements) {
+    throw new TypeError('Only one of `allowedElements` and `disallowedElements` should be defined')
+  }
+
+  return options.allowedElements || options.disallowedElements || options.allowElement
+    ? transform
+    : undefined
+
+  function transform(tree) {
+    visit(tree, 'element', onelement)
+  }
+
+  function onelement(node, index, parent) {
+    let remove
+
+    if (options.allowedElements) {
+      remove = !options.allowedElements.includes(node.tagName)
+    } else if (options.disallowedElements) {
+      remove = options.disallowedElements.includes(node.tagName)
+    }
+
+    if (!remove && options.allowElement) {
+      remove = !options.allowElement(node, index, parent)
+    }
```

Comment:
Is/was this covered by line 23-25?
Or does this check more?

Reply:
Which lines where?

`allowElement` is a function. `allowedElements` an array of strings. So yeah, this does do more!